client/allocrunner/taskrunner/task_runner.go

```diff
@@ -462,8 +462,29 @@ func (tr *TaskRunner) runDriver() error {
 	// TODO(nickethier): make sure this uses alloc.AllocatedResources once #4750 is rebased
 	taskConfig := tr.buildTaskConfig()
 
-	// TODO: load variables
+	// Build hcl context variables
+	vars, errs, err := tr.envBuilder.Build().AllValues()
+	if err != nil {
+		return err
+	}
+
+	// Handle per-key errors
+	if len(errs) > 0 {
+		keys := make([]string, 0, len(errs))
+		for k, err := range errs {
+			keys = append(keys, k)
+
+			if tr.logger.IsTrace() {
+				// Verbosely log every diagnostic for debugging
+				tr.logger.Trace("error building environment variables", "key", k, "error", err)
+			}
+		}
+
+		tr.logger.Warn("some environment variables not available for rendering", "keys", strings.Join(keys, ", "))
```

Comment:
Should this be a task event? It seems like this should be something surfaced to the user.

Reply:
I think it may become noisy if they actually need to expose a variable with invalid dots to an app. It would be a task event that is unactionable forever